### PR TITLE
04 array zamonier

### PR DESCRIPTION
**Pull request by zamonier**

### CODE REVIEW

File: src/main/java/array/Array1d.java [hunk 1]
```diff
@@ -10,40 +11,91 @@
...
 
     @Override
     public void visit(Consumer<T> consumer) {
-        throw new UnsupportedOperationException();
+        for (Object o : array) {
+            consumer.accept((T) o);
+        }
+    }
+
+    private static int valid(int size) {
```
Comment (PaulFrmBrn) [inline src/main/java/array/Array1d.java:76]: it's not a noun
validate?

File: src/main/java/array/Array1d.java [hunk 2]
```diff
@@ -10,40 +11,91 @@
...
+        if (size < 0) {
+            throw new NegativeArraySizeException();
+        }
+        return size;
+    }
+
+    private int valid(Coordinates1d coordinates) {
```
Comment (PaulFrmBrn) [inline src/main/java/array/Array1d.java:83]: +1

File: src/main/java/array/Array1d.java [hunk 3]
```diff
@@ -10,40 +11,91 @@
...
+            throw new IllegalArgumentException(Coordinates1d.class.getSimpleName() + " must be non-null");
+        }
+        var index = coordinates.getIndex();
+        if (index < 0 || index >= array.length) {
+            throw new ArrayIndexOutOfBoundsException(index);
+        }
+        return index;
+    }
+
+    private T valid(T value) {
```
Comment (PaulFrmBrn) [inline src/main/java/array/Array1d.java:94]: +1

File: src/main/java/array/Array1d.java (hunk 1 above)
Comment (zamonier) [reply src/main/java/array/Array1d.java:76]: It's adjective
```
var s = valid(input parameter)
```
```
var s = valid(size) 
```

**Pull request merged**